README.md

```diff
@@ -1,3 +1,22 @@
-# opm-common
-Contains common components used throughout all of OPM,
-in particular CMake modules for the build system.
+# opm-common: Tools for Eclipse reservoir simulation files
+
+The Open Porous Media (OPM) software suite provides libraries and
+tools for modeling and simulation of porous media processes,
+especially  for simulating CO2 sequestration and improved and enhanced
+oil recovery. The Eclipse file format is widely used in the reservoir
+simulation  community. This package provides a library containing code
+for processing files in Eclipse format, Python binding for accessing
+Eclipse files and  utility code used by other OPM modules (e.g. CMake
+modules used).
+
+License
+-------
+
+The library is distributed under the GNU General Public License,
+version 3 or later (GPLv3+).
+
+Requirements
+-----------
+
+- C++17 compatible compiler
+- CMake >= 3.10
```

Comment:
We **may** want to add "Posix compatibility" here since we do use Posix functions, notably `fnmatch`, in a number of locations.

Reply:
Lets wait for #2925

Reply:
> Let's wait for #2925

PR #2925 definitely helps, but doesn't fully resolve the issue.  We use `fileno()` and `isatty()` from `<unistd.h>` as part of our [logging](https://github.com/OPM/opm-common/blob/24e58e36aab0edfc1094b0725c1300e017a8f3b7/src/opm/common/OpmLog/OpmLog.cpp#L43) system (colourised output).  We could **probably** push that functionality into libfmt (using function `fmt::print()`), but that would require quite a bit of work in our logger, and also prevent fully switching to `std::format()` if we upgrade to C++20 at a later date.  We also have a few unit tests that use `<unistd.h>` and even [`<sys/*.h>`](https://github.com/OPM/opm-common/blob/24e58e36aab0edfc1094b0725c1300e017a8f3b7/tests/parser/ParseContext_EXIT1.cpp#L3-L4), but these can obviously be disabled if a user does not want them.

Reply:
One more case that requires Posix compatibility: We have a number of example and testing utilities, e.g.,

* [`convertECL`](https://github.com/OPM/opm-common/blob/089b6bff42fffa4b3291591f5779b521d9717b52/test_util/convertECL.cpp#L133)
* [`compareECL`](https://github.com/OPM/opm-common/blob/089b6bff42fffa4b3291591f5779b521d9717b52/test_util/compareECL.cpp#L99)
* [`opmpack`](https://github.com/OPM/opm-common/blob/089b6bff42fffa4b3291591f5779b521d9717b52/examples/opmpack.cpp#L110)

that all use `getopt()` from `<unistd.h>` (or `<getopt.h>`).

Reply:
added